pilot/pkg/networking/core/v1alpha3/listener.go

```diff
@@ -165,6 +165,7 @@ var (
 			// If client sends mTLS traffic, transport protocol will be set by the TLS inspector
 			TransportProtocol: xdsfilters.TLSTransportProtocol,
 			Protocol:          istionetworking.ListenerProtocolTCP,
+			MTLS:              true,
```

Comment:
Would it be better if we hold model.MutualTLSMode here - for any future uses?

Reply:
each list is for a specific mode. within that list , each entry either is for a tls type or raw_buffer. So I am not sure this should be mtls mode, as it's really a boolean at this point. It could be called "needsTlsContext" though?

Reply:
Yeah. Sorry When I went through the rest of the PR also realized mtls mode does not suit here.  I am fine leaving it as is .